std/src/main/scala/calico/std/History.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2022 Arman Bilge
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package calico.std
+
+import cats.effect.kernel.Async
+import cats.effect.kernel.Resource
+import cats.effect.std.Dispatcher
+import cats.syntax.all.*
+import fs2.Stream
+import fs2.concurrent.Channel
+import fs2.concurrent.Signal
+import io.circe.Decoder
+import io.circe.Encoder
+import io.circe.scalajs.*
+import org.scalajs.dom.EventListenerOptions
+import org.scalajs.dom.PopStateEvent
+import org.scalajs.dom.URL
+import org.scalajs.dom.window
+
+import scala.scalajs.js
+
+sealed trait History[F[_], S]:
+
+  def state: Signal[F, S]
+  def length: Signal[F, Int]
+
+  def forward: F[Unit]
+  def back: F[Unit]
+  def go: F[Unit]
+  def go(delta: Int): F[Unit]
+
+  def pushState(state: S): F[Unit]
+  def pushState(state: S, url: URL): F[Unit]
+
+  def replaceState(state: S): F[Unit]
+  def replaceState(state: S, url: URL): F[Unit]
+
+object History:
+  def make[F[_], S: Decoder: Encoder](using F: Async[F]): Resource[F, History[F, S]] =
+    Dispatcher[F].map { dispatcher =>
+      new:
```

Comment:
What does this syntax do?

Reply:
The `new:` ? This is Scala 3 shorthand for `new History[F, S] { ... }`